### DISCUSSION TITLE
issue: handleSubmit does not throw error when validation is invalid

### DISCUSSION THREAD

**Discussion by misako0927** [react-hook-form/react-hook-form#11614] [Q&A]
### Version Number

7.51.0

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/handlesubmit-does-not-throw-error-when-validation-is-invalid-h5tfmj?file=%2Fsrc%2FApp.tsx

### Steps to reproduce

1. Go to CodeSandbox
2. Click on submit

### Expected behaviour

expect: isSuccesSubmit set to false if validation has error
actual: isSuccessSubmit set to true regardless validation result

I think it should be checked there has validation errors in below logic:
https://github.com/react-hook-form/react-hook-form/blob/97c3ba46b9d34245a9a8c9febc953809f667944b/src/logic/createFormControl.ts#L1162

I expected that result but it might be misunderstanding.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
Sandbox not found.

**Comment by misako0927** [comment]
@Moshyfawn I'm sorry. I changed directory. Could you see this?:
https://codesandbox.io/p/sandbox/handlesubmit-does-not-throw-error-when-validation-is-invalid-h5tfmj?file=%2Fsrc%2FApp.tsx

**Comment by Moshyfawn** [comment]
No, still "not found" 😅

**Comment by misako0927** [comment]
I changed permission. I'm sorry for inconvenience.

**Comment by Moshyfawn** [comment]
You are calling `handleSubmit` inside the `try catch` block, which shouldn't throw. It's the `onValid` callback inside `handleSubmit` that throws in case there was an error in your submit handler.

You can refer to [the rules section](https://react-hook-form.com/docs/useform/handlesubmit) for a `try catch` example.

**Comment by misako0927** [comment]
Thank you for your comment and I'm sorry for late reply.
I understand thrown error which is in `handleSubmit` is only used for `onValid`.
